CERTZ2.1 server_certificates

priyacj opened the pull request:
CERTZ2.1 server_certificates testcase
CERTZ2.2 server_certificates testcase

priyacj committed:
Added CERTZ2.1 testcases

priyacj committed:
Updated README.md under server_certificates

priyacj committed:
modified README

priyacj committed:
fixed rundata

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+	}
+	cmd.Dir = dirPath
+	fmt.Printf("Executing cleanup command")
+	err := cmd.Start()
```
```go
if err := cmd.Start(); err != nil { ... }
`

priyacj replied:
Updated in the latest version.

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+		fmt.Printf("unable to run testdata cleanup command:%v", err)
+		return err
+	}
+	err = cmd.Wait()
```
Same as above.

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+}
+
+// PostValidationCheck function to do a validation of all services after certz rotation.
+func PostValidationCheck(t *testing.T, caCert *x509.CertPool, ctx context.Context, san, serverAddr, username, password string, cert tls.Certificate) bool {
```
Same comment re: `context.Context` here.

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+	return true
+}
+
+// PreInitCheck function to do a validation of gNMI/gNOI/gRIBI/p4RT services before certz rotation.
```
Can you explain why there is a difference here between the `PreInitCheck` and the `PostValidationCheck`? It seems like we need to have there is some opportunity here to have some commonality -- given that we'd expect that all services connect before and after.

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+	return true
+}
+
+// MismatchPostValidationCheck function to do a validation of all services after certz rotation.
```
Can you perhaps document more clearly what the differences between the different post validation checks here are? It looks like you might be able to have something where we could just hand an argument that asks for the expected state, rather than code repetition here?

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+		t.Errorf("FAIL: new gNMI Connection worked with mismatch trustbundle: got %v, want false", result)
+	}
+	t.Logf("PASS:New gNMI connection with mismatch trustbundle is not working as expected- got %v, want false.", result)
+	return true
```
This function always returns true -- is it ever meant to return false?

priyacj replied:
On validation failures of the services , it would error out .

robshakir commented on feature/security/gnsi/certz/tests/server_certificates/server_certificates_test.go:
```diff
@@ -0,0 +1,240 @@
+// TestServerCertTcTwo tests the server certificates from a set of one CA are able to be validated and
+// used for authentication to a device when used by a client connecting to each
+// gRPC service.
+
```
Remove blank line.

priyacj replied:
blank line is removed.

robshakir commented on feature/security/gnsi/certz/tests/server_certificates/server_certificates_test.go:
```diff
@@ -0,0 +1,240 @@
+			}
+			batch.Set(t, dut)
+			t.Logf("%s:replaced gNMI config with new ssl profile successfully.", tc.desc)
+			time.Sleep(10 * time.Second)
```
Can we add a comment that explains why we sleep here please?

priyacj replied:
removed the sleep wait time.

robshakir commented:
Thanks for this test case -- it looks to cover the `Certz` tests explained. I have a few questions and Go implementation comments.

priyacj committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
updated the README file

priyacj committed:
fixed permisison issue

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+		}
+		trust = append(trust, p)
+	}
+	if len(trust) > 0 {
```
done .

priyacj committed:
fixed few givencomments

priyacj committed:
fixed the executable bit for this file

priyacj committed:
added comments

priyacj committed:
added few comment lines

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

robshakir commented:
/gcbrun

priyacj committed:
Merge branch 'main' into certz2.1

priyacj committed:
added certz2.2 testcase

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+	return &certzpb.CertificateChain{}
+}
+
+// CertzRotate function to request the client certificate rotation.
```
done

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,641 @@
+	"encoding/pem"
+	"fmt"
+
+	//"io"
```
Remove stale unused import?

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,641 @@
+	*creds.UserPass
+}
+
+func (r *rpcCredentials) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
```
```suggestion
func (r *rpcCredentials) GetRequestMetadata(_ context.Context, _ ...string) (map[string]string, error) {
```

priyacj replied:
fixed it in the latest.

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,641 @@
+	//a valid check for trust not empty
+	if len(trust) == 0 {
+		return &certzpb.CertificateChain{}
+	} else {
```
Per the static check, you can just change this to:

```go

if len(trust) == 0 {
  return &certzpb.CertificateChain{}
}

<rest of code>
```

since the `else` is not adding any new condition.

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,641 @@
+}
+
+// CertzRotate function to request the client certificate rotation and returns true on successful rotation.
+func CertzRotate(t *testing.T, caCert *x509.CertPool, certzClient certzpb.CertzClient, cert tls.Certificate, san, serverAddr, profileID string, entities ...*certzpb.Entity) bool {
```
There are unused arguments for this function -- since this isn't meeting a signature defined elsewhere, please remove the arguments.

priyacj replied:
fixed it in the latest. please check

robshakir commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+	}
+	rotateResponse := &certzpb.RotateCertificateResponse{}
+	for i := 0; i < 20; i++ {
+		rotateResponse, err = rotateRequestClient.Recv()
```
Did this comment make sense, please let me know if you'd like to discuss it further.

priyacj replied:
can you please elaborate more on this ?

priyacj replied:
fixed on the latest . please check.

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
fixed deprecated grpc.WithBlock:DialOption

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
fixed all the comments

priyacj committed:
fixed the else part

priyacj committed:
fixed the staticerror

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

morrowc reviewed (commented):
please address static-analysis issues.

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
feedback comments are checkedin

priyacj committed:
fixed the lints

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

morrowc commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,586 @@
+	*creds.UserPass
+}
+
+func (r *rpcCredentials) GetRequestMetadata(_ context.Context, _ ...string) (map[string]string, error) {
```
it's been a while and I may have asked this previously, but why does this function need to accept a context.Context which it never uses?

Is this mocking some production version of this function?
This looks like a whole new package, and the tests that appear to use this just use this directly by name, why add to the function signature something that just isn't used?

morrowc replied:
no reply here, what's happening with this?

morrowc commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,586 @@
+package setupservice
+
+import (
+	context "context"
```
Aliasing this to itself isn't good. please remove alias.
(this only appears 1x in the codebase today, on: ./internal/cntrsrv/proto/cntr/cntr_grpc.pb.go 
which I believe is a generated file, so we should follow the normal style/standard here and not alias context in this file/package)

morrowc replied:
you didn't seem to fix this.

morrowc commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,586 @@
+	EntityTypeAuthPolicy entityType = 3
+)
+
+// CertificateChainRequest is an input argument for the  type definition for the  CreateCertzChain.
```
since this struct definition is for the function after the function below, can you move it down there?

I think either all of this gets defined at the top of the file, or it gets defined 'right next to' the thing that requires it.
randomly in the middle of the file isn't helping readers and 'future you', because: "Wait, why is this here? is it important? <scroll, scroll, scroll... nope it's just randomly here? huh!>"

morrowc commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,586 @@
+		}
+		trust = append(trust, p)
+	}
+	//a valid check for trust not empty
```
// a valid c...

morrowc replied:
almost certainly ALSO make this a proper / full sentence, not a sentence fragment.

morrowc commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,586 @@
+	}
+	var prevCert *certzpb.CertificateChain
+	var bundleToReturn *certzpb.CertificateChain
+	for i := len(trust) - 1; i >= 0; i-- {
```
If it helps here, the trustbundle isn't really a 'certificate-chain'.
Or I don't expect it actually is a chain, it's just a list of root certificates/public-keys. There's no parent-child relationship REALLY between the various certs in the bundle.

You don't need to roll them up into a chain in any particular order, you could avoid some:
  for i = 0  ; i < ....
  
  business with just:
    for _, cert := range trust{}

if you wanted.

morrowc replied:
this would also avoid you indexing into the array, which may make this a tad faster ... that isn't a huge concern I suppose :) but it's nice :)

morrowc commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,586 @@
+		}
+	}
+	return bundleToReturn
+
```
extra newline?

morrowc reviewed (commented):
I'll probably have more comments, i got bogged down a touch :(

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
fixed the feedback comments

priyacj committed:
incorporated the feedback

priyacj committed:
fixed the executable bit

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

morrowc commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+	rotateCertRequest := &certzpb.RotateCertificateRequest{
+		ForceOverwrite: false,
+		SslProfileId:   profileID,
+		RotateRequest:  rotateRequest}
```
this doesn't look like it got fixed?

morrowc reviewed (commented):
There's a bunch of 'not attended to' questions/comments in this.

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

mihirpitale-googler commented:
hello @priyacj , can you address comments and please resolve them?

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
addressed comments

priyacj committed:
added comment READMEfile

priyacj committed:
fixed go.mod conflict error

priyacj committed:
removed commented line#10 from mk_cash.sh

priyacj committed:
updated gofile

priyacj committed:
Update README.md

updated canonical OC

priyacj committed:
Update README.md with canonical OC

priyacj committed:
Update server_certificates_test.go

priyacj committed:
fixed typo errors

priyacj committed:
Update README.md

priyacj committed:
Update README.md

priyacj committed:
fixed go file

priyacj committed:
Update README.md

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Update setup_service.go

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Update README.md for the canonical errors

priyacj committed:
Update README.md with name field

priyacj committed:
Update README.md with error

priyacj committed:
Update README.md fix-1

priyacj committed:
newadd for certz2.1

priyacj committed:
moved the files to featureprofiles/feature/gnsi/ from featureprofiles/feature/security/gnsi

priyacj committed:
fixed the go package

priyacj committed:
fixed metadata CERTZ-2

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
updated server_certificates_test.go

priyacj committed:
Update setup_service.go

priyacj committed:
Update setup_service.go

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+}
+
+// ServerCertzRotate function to request the server certificate rotation.
+func ServerCertzRotate(t *testing.T, caCert *x509.CertPool, certzClient certzpb.CertzClient, cert tls.Certificate, ctx context.Context, dut *ondatra.DUTDevice, san, serverAddr, profileID string, entities ...*certzpb.Entity) bool {
```
Have updated the code to use the same library function CertzRotate for every rotate RPC call . Please check.
// CertzRotate function to request the server certificate rotation and returns true/false on successful/failed rotation.
func CertzRotate(ctx context.Context, t *testing.T, newcaCert *x509.CertPool, certzClient certzpb.CertzClient, gnmiClient gnmipb.GNMIClient, newclientCert tls.Certificate, dut *ondatra.DUTDevice, username string, password string, san, serverAddr, profileID string, newTLS bool, mismatch bool, scale bool, entities ...*certzpb.Entity) bool {

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+		if success {
+			break
+		}
+		if i != 10 {
```
I have removed the condition check as it will not be hit.

priyacj committed:
Update setup_service.go

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+		if err != nil {
+			t.Fatalf("Error sending rotate finalize request: %v", err)
+		}
+		err = rotateRequestClient.CloseSend()
```
updated the code as per the given input in the latest version.

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+// CertCleanup function to  clean out the CA content under test_data.
+func CertCleanup(dirPath string) error {
+	cmd := exec.Cmd{
+		Path:   "./cleanup.sh",
```
can you please elaborate more on this ?

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+}
+
+// VerifyGnmi function to validate the gNMI service RPC after successful rotation.
+func VerifyGnmi(t *testing.T, caCert *x509.CertPool, ctx context.Context, san, serverAddr, username, password string, cert tls.Certificate) bool {
```
removed the context in the latest version.

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+}
+
+// VerifyGnsi function to validate the gNSI service RPC after successful rotation.
+func VerifyGnsi(t *testing.T, caCert *x509.CertPool, ctx context.Context, san, serverAddr, username, password string, cert tls.Certificate) bool {
```
removed passing context as an argument in the latest version. please check.

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+	rotateCertRequest = &certzpb.RotateCertificateRequest{
+		ForceOverwrite: false,
+		SslProfileId:   profileID,
+		RotateRequest:  finalizeRequest}
```
Thanks for the input. I have updated the formatting in the latest code.

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+}
+
+// ServerCertzRotate function to request the server certificate rotation.
+func ServerCertzRotate(t *testing.T, caCert *x509.CertPool, certzClient certzpb.CertzClient, cert tls.Certificate, ctx context.Context, dut *ondatra.DUTDevice, san, serverAddr, profileID string, entities ...*certzpb.Entity) bool {
```
Taken care of context.Context the first argument of this function and docstrings for the arguments in the main go.Please check.

priyacj commented on feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go:
```diff
@@ -0,0 +1,750 @@
+}
+
+// CertGeneration function to create test data for use in TLS tests.
+func CertGeneration(dirPath string) error {
```
We have implemented test data generation using a shell script based on the README instructions. Please let us know if you would prefer it to be implemented in Go instead. Keeping the test data pre-generated and passing it to script as an argument is another approach as dynamic generation (via shell script or Go) for 20,000 CAs takes significantly longer (~45 min for each set (rsa, ecdsa)). Please share your thoughts.

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
updated the given feedback from 2.1/2.2

priyacj committed:
Merge branch 'main' into certz2.1

priyacj committed:
Update server_certificates_test.go

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Update server_certificates_test.go -fmt errors

priyacj committed:
fixed the seen fmt error

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

mihirpitale-googler reviewed (approved):
lgtm. thanks for clarification/rectification. priya.

lvaish05 committed:
Merge branch 'main' into certz2.1

priyacj committed:
fixed the go,mod

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

morrowc commented on feature/gnsi/certz/test_data/mk_cas.sh:
```diff
@@ -5,7 +5,8 @@
 # in each directory.
-DIRS=(01 02 10 1000 20000)
+#DIRS=(01 02 10 1000 20000)
+DIRS=(01 02 10 1000)
```
the 20k set is actually important for performance testing.
I feel like i've said this before.

dplore replied:
thanks!

dplore replied:
marking resolved

morrowc committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

morrowc committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'main' into certz2.1

priyacj committed:
Add '20000' to cleanup script directory loop

priyacj committed:
Add '20000' to CA directories list

priyacj committed:
Clean up TODO comments in README.md

Removed TODO comments regarding trust_bundle configurations and CA permutations.

priyacj committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

lvaish05 committed:
Merge branch 'main' into certz2.1

dplore commented:
@priyacj please provide logs for the test results

dplore commented:
/fptest virtual

dplore commented:
Test is failing on ncptx container.  
Step #1 - "fp-presubmit": *** PROPERTY: dut.os_version -> 22.3X80-D49.64-EVO


See [logs](https://ff4cd3bc89309b539aef015e06bb5b86398275e369241afe1bf2a83-apidata.googleusercontent.com/download/storage/v1/b/featureprofiles-ci-logs-juniper/o/log-57535e79-9825-44a9-9567-e3e3a7716320.txt?jk=AXH35DE6xoN_iYloXOe5ya3y4UbIe6od-e6Ky8AisMvaw5nLpKdTeVR5dLg7xiliRM5_Buzxdh8USmpntCTeievPSMqTsCjcKsiqbBQqd2d5Xg0orUMIJAv7VA-tjhfaTBw7m18BWONlXM7as6g_rZstEQkSlTcZp3HBM1L8FFNSWftXMPgZkQIGOSOHJOZ_cKZfIOoMqzOLhOUyMp6f_djOxrAlRfQAqZVjfj4d-lpZiSS-6XVEbF2l1a1FPNOYBG9gREJ5oJa1c8ZKG-1l4tx9OZoXJp_9VoKL_K7xRknmduLytbOLW9daKBa6QMZ85rrAfAZndKutcJ34vKsEC8emmYQAP210DcunxGE7F8x2VEW_3YYTcWo73sv7BNip5GZJllaiLfYn0rVmU1mGD3hB9kk817gociymLp2Zn11K5Hvxy74cju8eJ7wfb5-Jp_mnrZ4tO2XT8ju7UQ2g1Pf081dqbtdDg24rOlqSxY9cMA1N6iz72Ojv0gMNNCVdRI_0B6izptx64mco2L5S2q-kEIDI30dee22xLEHj626aCfW1EYQY2_C6531BcZ7dbWW1BEH_D27VZ5FL5Ypmd_D3GMb2fHtdrsIHegXQQhWVWOW69Rn1rbJzdfPR9KU8PyvYr2nGX-Wa6sKlp_HMqT745Ier8HageU9-SXibcJSjc9AsSNXBxCtltuhEFW3qJZhTMRYHxP0l51NFAJ7Vbq-csmFTcMHvLS_tUXLT7_Ohp-6i1XQt8NmrB2q3RHjiDNdwHcOPfNAdyOrFWIXw_ThxSfywocLCZ8WM9vY1bt63X440aijDMbmvFVlmb2up7l-dT60d8VqvUzRMbrvnf9UxIA_ss-xCRi1krewcZ1m2kK7Mtfg19wOYVua7w4CoVfkDJwKNOu7yV__IlDIG2HgnV9Kl5A6YUEJ5zT09pS5BIhkG1JppJ13CFKVnhATHDMS2kb8OFVULkVD-A5Cf9U6Y2vFP5xFT7bvr_96hHkYX8aQDUuZUtmJur2rYBwMPIJiTGajmGAj3OKxKzBZ9KsBEdhKGOsbhCOTwaNibeoX-lH9YbzbpXFtJmyztgO2sMm4fzR_FZOlWxDJhDX7WxdtYIFudsPcgFrL0ZVlYcYskJoetKuw15nzBPJM7eRrnyRzhRUNsoaTOcsjEYfIwt9Cm6oZwqIRP0YCYoTKijQI-NHwjeCqxHu3WUH9_EczajPapPPj9laBHgtzP9M2Bi-FZ79Ie&isca=1https://ff4cd3bc89309b539aef015e06bb5b86398275e369241afe1bf2a83-apidata.googleusercontent.com/download/storage/v1/b/featureprofiles-ci-logs-juniper/o/log-57535e79-9825-44a9-9567-e3e3a7716320.txt?jk=AXH35DE6xoN_iYloXOe5ya3y4UbIe6od-e6Ky8AisMvaw5nLpKdTeVR5dLg7xiliRM5_Buzxdh8USmpntCTeievPSMqTsCjcKsiqbBQqd2d5Xg0orUMIJAv7VA-tjhfaTBw7m18BWONlXM7as6g_rZstEQkSlTcZp3HBM1L8FFNSWftXMPgZkQIGOSOHJOZ_cKZfIOoMqzOLhOUyMp6f_djOxrAlRfQAqZVjfj4d-lpZiSS-6XVEbF2l1a1FPNOYBG9gREJ5oJa1c8ZKG-1l4tx9OZoXJp_9VoKL_K7xRknmduLytbOLW9daKBa6QMZ85rrAfAZndKutcJ34vKsEC8emmYQAP210DcunxGE7F8x2VEW_3YYTcWo73sv7BNip5GZJllaiLfYn0rVmU1mGD3hB9kk817gociymLp2Zn11K5Hvxy74cju8eJ7wfb5-Jp_mnrZ4tO2XT8ju7UQ2g1Pf081dqbtdDg24rOlqSxY9cMA1N6iz72Ojv0gMNNCVdRI_0B6izptx64mco2L5S2q-kEIDI30dee22xLEHj626aCfW1EYQY2_C6531BcZ7dbWW1BEH_D27VZ5FL5Ypmd_D3GMb2fHtdrsIHegXQQhWVWOW69Rn1rbJzdfPR9KU8PyvYr2nGX-Wa6sKlp_HMqT745Ier8HageU9-SXibcJSjc9AsSNXBxCtltuhEFW3qJZhTMRYHxP0l51NFAJ7Vbq-csmFTcMHvLS_tUXLT7_Ohp-6i1XQt8NmrB2q3RHjiDNdwHcOPfNAdyOrFWIXw_ThxSfywocLCZ8WM9vY1bt63X440aijDMbmvFVlmb2up7l-dT60d8VqvUzRMbrvnf9UxIA_ss-xCRi1krewcZ1m2kK7Mtfg19wOYVua7w4CoVfkDJwKNOu7yV__IlDIG2HgnV9Kl5A6YUEJ5zT09pS5BIhkG1JppJ13CFKVnhATHDMS2kb8OFVULkVD-A5Cf9U6Y2vFP5xFT7bvr_96hHkYX8aQDUuZUtmJur2rYBwMPIJiTGajmGAj3OKxKzBZ9KsBEdhKGOsbhCOTwaNibeoX-lH9YbzbpXFtJmyztgO2sMm4fzR_FZOlWxDJhDX7WxdtYIFudsPcgFrL0ZVlYcYskJoetKuw15nzBPJM7eRrnyRzhRUNsoaTOcsjEYfIwt9Cm6oZwqIRP0YCYoTKijQI-NHwjeCqxHu3WUH9_EczajPapPPj9laBHgtzP9M2Bi-FZ79Ie&isca=1https://ff4cd3bc89309b539aef015e06bb5b86398275e369241afe1bf2a83-apidata.googleusercontent.com/download/storage/v1/b/featureprofiles-ci-logs-juniper/o/log-57535e79-9825-44a9-9567-e3e3a7716320.txt?jk=AXH35DE6xoN_iYloXOe5ya3y4UbIe6od-e6Ky8AisMvaw5nLpKdTeVR5dLg7xiliRM5_Buzxdh8USmpntCTeievPSMqTsCjcKsiqbBQqd2d5Xg0orUMIJAv7VA-tjhfaTBw7m18BWONlXM7as6g_rZstEQkSlTcZp3HBM1L8FFNSWftXMPgZkQIGOSOHJOZ_cKZfIOoMqzOLhOUyMp6f_djOxrAlRfQAqZVjfj4d-lpZiSS-6XVEbF2l1a1FPNOYBG9gREJ5oJa1c8ZKG-1l4tx9OZoXJp_9VoKL_K7xRknmduLytbOLW9daKBa6QMZ85rrAfAZndKutcJ34vKsEC8emmYQAP210DcunxGE7F8x2VEW_3YYTcWo73sv7BNip5GZJllaiLfYn0rVmU1mGD3hB9kk817gociymLp2Zn11K5Hvxy74cju8eJ7wfb5-Jp_mnrZ4tO2XT8ju7UQ2g1Pf081dqbtdDg24rOlqSxY9cMA1N6iz72Ojv0gMNNCVdRI_0B6izptx64mco2L5S2q-kEIDI30dee22xLEHj626aCfW1EYQY2_C6531BcZ7dbWW1BEH_D27VZ5FL5Ypmd_D3GMb2fHtdrsIHegXQQhWVWOW69Rn1rbJzdfPR9KU8PyvYr2nGX-Wa6sKlp_HMqT745Ier8HageU9-SXibcJSjc9AsSNXBxCtltuhEFW3qJZhTMRYHxP0l51NFAJ7Vbq-csmFTcMHvLS_tUXLT7_Ohp-6i1XQt8NmrB2q3RHjiDNdwHcOPfNAdyOrFWIXw_ThxSfywocLCZ8WM9vY1bt63X440aijDMbmvFVlmb2up7l-dT60d8VqvUzRMbrvnf9UxIA_ss-xCRi1krewcZ1m2kK7Mtfg19wOYVua7w4CoVfkDJwKNOu7yV__IlDIG2HgnV9Kl5A6YUEJ5zT09pS5BIhkG1JppJ13CFKVnhATHDMS2kb8OFVULkVD-A5Cf9U6Y2vFP5xFT7bvr_96hHkYX8aQDUuZUtmJur2rYBwMPIJiTGajmGAj3OKxKzBZ9KsBEdhKGOsbhCOTwaNibeoX-lH9YbzbpXFtJmyztgO2sMm4fzR_FZOlWxDJhDX7WxdtYIFudsPcgFrL0ZVlYcYskJoetKuw15nzBPJM7eRrnyRzhRUNsoaTOcsjEYfIwt9Cm6oZwqIRP0YCYoTKijQI-NHwjeCqxHu3WUH9_EczajPapPPj9laBHgtzP9M2Bi-FZ79Ie&isca=1)

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1

dplore committed:
Merge branch 'main' into certz2.1

sudhirakondi commented:
Task Created to track running Ondatra tests on the ncptx:
https://partnerissuetracker.corp.google.com/issues/507392581 

Also, attached pass logs for the certz tests to above.

dplore committed:
Merge branch 'main' into certz2.1

priyacj committed:
Merge branch 'openconfig:main' into certz2.1
